packages/material-ui-lab/src/SelectableGroup/SelectableGroup.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as utils from './utils';
+import SelectableGroupContext from './SelectableGroupContext';
+
+function SelectableGroup(props) {
+  const [selected, setSelected] = React.useState(null);
+  const { additional, children, exclusive, value: valueProp } = props;
+
+  const createReducer = handler => {
+    return (event, value) => {
+      setSelected(prevSelected => {
+        const newSelected = handler(prevSelected, value, exclusive);
+
+        // Only call onChange if it exists and state has changed (Object.is mimics React)
+        if (props.onChange && !Object.is(newSelected, prevSelected)) {
+          props.onChange(event, newSelected);
+        }
+
+        return newSelected;
+      });
+    };
+  };
+
+  const deselect = createReducer(utils.deselect);
+  const select = createReducer(utils.select);
+  const toggle = createReducer(utils.toggle);
+  const isValueSelected = value => utils.hasValue(selected, value);
+
+  React.useEffect(() => {
```

Comment:
Why do we need useEffect? There is no SSR support and we have to render twice to get the initial value.